### PR TITLE
DT-2222  Add queue to smoke test to test for release/recall events produced

### PR DESCRIPTION
**Pull request by karenmillermoj**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/dpssmoketest/service/QueueService.kt [hunk 1]
```diff
@@ -39,10 +61,42 @@ class QueueService(
...
+      ).messages.firstOrNull()
+        ?.also { msg ->
+          val (message, messageId, messageAttributes) = objectMapper.readValue(msg.body, HMPPSMessage::class.java)
+          val eventType = messageAttributes.eventType.Value
+          val hmppsDomainEvent = objectMapper.readValue(message, HMPPSDomainEvent::class.java)
+
+          log.info("Received message $message $messageId type $eventType")
+          hmppsEventQueueSqsClient.deleteMessage(DeleteMessageRequest(hmppsEventQueueUrl, msg.receiptHandle))
+          if (eventTypeRequired == eventType && hmppsDomainEvent.additionalInformation.nomsNumber == nomsNumber) eventFound = true
+        }
```
Comment (karenmillermoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/dpssmoketest/service/QueueService.kt:78]: Looking if there's a neater kotlin way to prettify this.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/dpssmoketest/service/QueueService.kt [hunk 2]
```diff
@@ -39,10 +61,42 @@ class QueueService(
...
   fun checkForEvent(eventTypeRequired: String, nomsNumber: String): Boolean {
-    return true
+    var eventFound = false
+    while (hmppsEventQueueSqsClient.countMessagesOnQueue()> 0 && !eventFound) {
+      hmppsEventQueueSqsClient.receiveMessage(
+        ReceiveMessageRequest(hmppsEventQueueUrl)
+          .withMaxNumberOfMessages(1)
+      ).messages.firstOrNull()
+        ?.also { msg ->
+          val (message, messageId, messageAttributes) = objectMapper.readValue(msg.body, HMPPSMessage::class.java)
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/dpssmoketest/service/QueueService.kt:71]: Maybe extract these lines to a extension function so you can write `msg.toHMPPSDOmainEvent()`

**Review by andymarke — APPROVED** [review]
LGTM

File: src/main/kotlin/uk/gov/justice/digital/hmpps/dpssmoketest/service/QueueService.kt (hunk 2 above)
Comment (karenmillermoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/dpssmoketest/service/QueueService.kt:71]: good idea - will look at that in next PR

**Pull request merged**